wallet/src/signer/software_signer/mod.rs

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common::chain::{
+    signature::{
+        inputsig::{
+            arbitrary_message::ArbitraryMessageSignature,
+            classical_multisig::authorize_classical_multisig::{
+                sign_classical_multisig_spending, AuthorizedClassicalMultisigSpend,
+                ClassicalMultisigCompletionStatus,
+            },
+            standard_signature::StandardInputSignature,
+            InputWitness,
+        },
+        sighash::{sighashtype::SigHashType, signature_hash},
+        DestinationSigError,
+    },
+    ChainConfig, Destination, Transaction, TxOutput,
+};
+use crypto::key::{
+    extended::{ExtendedPrivateKey, ExtendedPublicKey},
+    hdkd::{derivable::Derivable, u31::U31},
+    PrivateKey,
+};
+use itertools::Itertools;
+use randomness::make_true_rng;
+use serialization::Encode;
+use wallet_storage::WalletStorageReadUnlocked;
+use wallet_types::signature_status::SignatureStatus;
+
+use crate::{
+    account::PartiallySignedTransaction,
+    key_chain::{make_account_path, AccountKeyChains, FoundPubKey, MasterKeyChain},
+};
+
+use super::{Signer, SignerError, SignerResult};
+
+pub struct SoftwareSigner<'a, T: WalletStorageReadUnlocked> {
+    db_tx: &'a T,
+    chain_config: Arc<ChainConfig>,
```

Comment:
I get that this is for simplicity... but you already have a lifetime `'a`... why not use it for ChainConfig as well?

Reply:
Never mind. I think this is fine.